src/io/flutter/editor/ColorField.java

```diff
@@ -0,0 +1,224 @@
+    if (expression == null) return null;
+
+    final String colorsPrefix = "Colors.";
+    if (expression.startsWith(colorsPrefix)) {
+      final FlutterColors.FlutterColor flutterColor = FlutterColors.getColor(expression.substring(colorsPrefix.length()));
+      if (flutterColor != null) {
+        return flutterColor.getAWTColor();
+      }
+    }
+    return ExpressionParsingUtils.parseColor(expression);
+  }
+
+  private static String buildColorExpression(Color color) {
+    final String flutterColorName = FlutterColors.getColorName(color);
+    if (flutterColorName != null) {
+      // TODO(jacobr): only apply this conversion if the material library is
+      // already imported in the library being edited.
+      // We also need to be able to handle cases where the material library
+      // is imported with a prefix.
+      return "Colors." + flutterColorName;
+    }
+    final int alpha = color.getAlpha();
+    final int red = color.getRed();
+    final int green = color.getGreen();
+    final int blue = color.getBlue();
+    return String.format("Color(0x%02x%02x%02x%02x)", alpha, red, green, blue);
+  }
+
+  public void addTextFieldListeners(String name, JBTextField field) {
+    final FlutterOutline matchingOutline = panel.getCurrentOutline();
+    field.addActionListener(e -> panel.setPropertyValue(name, field.getText()));
+    field.addFocusListener(new FocusListener() {
+      @Override
+      public void focusGained(FocusEvent e) {
+        // The popup is gone if we have received the focus again.
+        disposeColorPicker();
+      }
+
+      @Override
+      public void focusLost(FocusEvent e) {
+        if (panel.getCurrentOutline() != matchingOutline) {
+          // Don't do anything. The user has moved on to a different outline node.
+          return;
+        }
+        if (e.isTemporary()) {
+          return;
+        }
+
+        if (colorPicker != null) {
+          // We lost focus due to the popup being opened so there is no need to
+          // update the value now. The popup will update the value when it is
+          // closed.
+          return;
+        }
+        panel.setPropertyValue(name, field.getText());
+      }
+    });
+  }
+
+  void cancelPopup() {
+    currentColor = colorAtPopupLaunch;
+    setText(expressionAtPopupLaunch);
+    panel.setPropertyValue(name, originalExpression, true);
+    repaint();
+    colorPicker.dispose();
+    colorPicker = null;
+  }
+
+  void disposeColorPicker() {
+    if (colorPicker != null) {
+      colorPicker.dispose();
+      colorPicker = null;
+    }
+  }
+
+  void showColorFieldPopup() {
+    disposeColorPicker();
+    colorPicker = ColorPickerFactory.create();
+    // The color picker is currently only supported in the AndroidStudio version of the plugin.
+    // TODO(jacobr): show a toast message explaining it is missing on IntelliJ for now.
+    if (colorPicker != null) {
+      colorAtPopupLaunch = currentColor;
```

Comment:
Can currentColor be passed to the show? and reset when colorPicker is disposed?  Since colorPicker is created each time?

Reply:
that is a bug. fixed to pass in currentColor.